homeassistant/components/tradfri/cover.py

```diff
@@ -0,0 +1,139 @@
+"""Support for IKEA Tradfri covers."""
+import logging
+
+from pytradfri.error import PytradfriError
+
+from homeassistant.components.cover import CoverDevice, ATTR_POSITION
+from homeassistant.core import callback
+from . import DOMAIN as TRADFRI_DOMAIN, KEY_API, KEY_GATEWAY
+from .const import CONF_GATEWAY_ID
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Load Tradfri covers based on a config entry."""
+    gateway_id = config_entry.data[CONF_GATEWAY_ID]
+    api = hass.data[KEY_API][config_entry.entry_id]
+    gateway = hass.data[KEY_GATEWAY][config_entry.entry_id]
+
+    devices_commands = await api(gateway.get_devices())
+    devices = await api(devices_commands)
+    covers = [dev for dev in devices if dev.has_blind_control]
+    if covers:
+        async_add_entities(TradfriCover(cover, api, gateway_id) for cover in covers)
+
+
+class TradfriCover(CoverDevice):
+    """The platform class required by Home Assistant."""
+
+    def __init__(self, cover, api, gateway_id):
+        """Initialize a cover."""
+        self._api = api
+        self._unique_id = f"{gateway_id}-{cover.id}"
+        self._cover = None
+        self._cover_control = None
+        self._cover_data = None
+        self._name = None
+        self._available = True
+        self._gateway_id = gateway_id
+
+        self._refresh(cover)
+
+    @property
+    def unique_id(self):
+        """Return unique ID for cover."""
+        return self._unique_id
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        info = self._cover.device_info
+
+        return {
+            "identifiers": {(TRADFRI_DOMAIN, self._cover.id)},
+            "name": self._name,
+            "manufacturer": info.manufacturer,
+            "model": info.model_number,
+            "sw_version": info.firmware_version,
+            "via_device": (TRADFRI_DOMAIN, self._gateway_id),
+        }
+
+    async def async_added_to_hass(self):
+        """Start thread when added to hass."""
+        self._async_start_observe()
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def should_poll(self):
+        """No polling needed for tradfri cover."""
+        return False
+
+    @property
+    def name(self):
+        """Return the display name of this cover."""
+        return self._name
+
+    @property
+    def current_cover_position(self):
+        """Return current position of cover.
+
+        None is unknown, 0 is closed, 100 is fully open.
```

Comment:
Same thoughts here, to turn around this. So 100 closed and 0 fully open.
When add the blinds to homekit, and drag the slider to 70% open you think the blinds close for 70%

![IMG_2516](https://user-images.githubusercontent.com/660835/65083237-48c34280-d9a8-11e9-84bc-be2fa2d1db08.PNG)